lib/similarity_utils.py

```diff
@@ -61,34 +61,48 @@ def prepare_pandas_row(pandas_row: pd.DataFrame) -> List[Any]:
     return p_list
 
 
-def similarity(actor1: List[Any], actor2: List[Any]) -> float:
+def similarity(actor1: List[Any], actor2: List[Any], reduced_dataset: bool = False) -> float:
     """metoda licząca similarity pomiędzy dwoma aktorami;
     jej argumentami są dwie listy, a wartością wyjściową wartość z przedziału [-1, 1];
+    parametr 'reduced_dataset' domyślnie ustawiony jest na False, jednak w przypadku gdy zbiór 
+    danych na którym liczymy similarity ograniczamy do listy poniższych 6 kolumn:
+    ["nconst", "tconst", "titleType", "genres", "category", "primaryName"]
+    to należy zmienić wartość parametru 'reduced_dataset' na True;
     metoda jest przygotowana pod dane ze zbioru JOINED_DATA"""
     weights = [0.3, 0.2, 0.3, 0.2]
-    values = [
-        iou(actor1[1], actor2[1]),  # similarity ze względu na ilość wspólnych filmów
-        iou(actor1[2], actor2[2]),  # similarity ze względu na rodzaj granych produkcji
-        iou(actor1[7], actor2[7]),  # similarity ze względu na gatunek granych produkcji
-        1 if actor1[8] == actor2[8] else 0  # similarity ze względu na tę samą płeć
-    ]
+    if reduced_dataset:
+        values = [
+            iou(actor1[1], actor2[1]),  # similarity ze względu na ilość wspólnych filmów
+            iou(actor1[2], actor2[2]),  # similarity ze względu na rodzaj granych produkcji
```

Comment:
chyba się nie zgadzam, jeśli wybieramy takie kolumny jak w opisie, to tu będzie wybierane titleType

Reply:
jak dla mnie wszystko w porządku. mamy listę
["nconst", "tconst", "titleType", "genres", "category", "primaryName"]
więc będzie to kolejno według indeksów
0 - nconst, 1 - tconst, 2 - titleType, 3 - genres, 4 - category, 5 - primaryName
1 - tconst odpowiada za zagranie w filmach;
2 - titleType za rodzaj granych produkcji;
3 - genres za gatunek produkcji;
4 - category za płeć aktora.